salt/states/pkgrepo.py

```diff
@@ -387,6 +402,14 @@ def absent(name, **kwargs):
         if not name.startswith('ppa:'):
             name = 'ppa:' + name
 
+    remove_key = any(kwargs.get(x) is not None
+                     for x in ('keyid', 'keyid_ppa'))
```

Comment:
No, the point of ``remove_key`` here is to see whether or not to run ``pkg.del_repo_key``. The logic in the ``del_repo_key()`` function will decide the behavior of whether or not the key is actually removed.